### DISCUSSION TITLE
Provide problem statement, security analysis and use cases

### DISCUSSION THREAD

**Discussion by Oskar-van-Deventer** [trustoverip/trust-spanning-protocol#15] [General]
At this moment, it remains unclear to me what problem the _TSP protocol_ aims to solve. The [indirectly referenced use cases from the  architecture spec](https://github.com/trustoverip/TechArch/blob/main/spec.md#5-example-use-cases) are about the higher ToIP layers, not about ToIP layer 2, where the TSP protocol resides. Moreover, we are talking about a security problem area, so I would expect a proper security analysis. [Section 8.3 on messages](https://github.com/trustoverip/TechArch/blob/main/spec.md#83-messages) provides four important security-related properties of the TSP protocol. However, these are not elaborated in terms of e.g. attack vectors or associated "negative" use cases, that would lack some of these properties. Also, I would expect that such properties/requirements do not only apply to the messages, but also other aspects of the protocol, like identification and routing.

Oskar

**Comment by talltree** [comment]
@Oskar-van-Deventer In terms of the problem the Trust Spanning Protocol aims to solve, I think the answer is in the name of this task force. It is true that the vast majority of the use cases we have discussed (which are still missing from the current public review draft of the [ToIP Technology Architecture Specification V1.0](https://github.com/trustoverip/TechArch/blob/main/spec.md) — that is a known action item) primarily involve trust tasks at Layer 3. But the primary purpose of a `trust spanning protocol` is to enable `trust tasks` to be performed across `trust domains`. So I would state that as the overall problem the TSP aims to solve.

In terms of security, there has never been any question that we'll want a proper security analysis of the TSP. In fact at the time we chartered this TF, it was agreed that ideally we would seek a formal method analysis of the protocol once we felt the spec was complete. (Phil Windley, who has a background in formal methods, has offered to help us find resources for this.)

Lastly, RE identification and routing, seven of the 18 ToIP Layer 2 requirements in the [ToIP Technology Architecture Specification V1.0](https://github.com/trustoverip/TechArch/blob/main/spec.md) are for ToIP identifiers. Routing was also discussed in the Intermediaries section, however if you feel specific requirements are missing there, please feel free to raise an issue as that spec is still in public review.
